### DISCUSSION TITLE
Large scale app: How best to struture livewire

### DISCUSSION THREAD

**Discussion by NBA707** [livewire/livewire#7442] [1. Help] [answered]
I have an app that I wrote and it covers several topics.   Each topic has perhaps up to 10 key areas themselves.   From Customers, Inventory, Work Orders, Vendor Management, Quote Management, Purchase Management.

What is the best way to orgranize them using Livewire?   would I just artisan livewire:make Major 3 Letter Acronym<dot>Minor 3 Letter Accronym<dot><Final 3 Letter Accryonym<dot>Topic?

artisian livewire:make Inv.Stk.Loc.Checker for example would be Inventory Stock Location Checker (there are 7 tocpics total)?

Or should it be slightly different and not really map easily to the Nav bar menu?

Thanks for any insight or thoughts on structuring my 646 blade files to make them easily findable.

**Comment by joshhanley** [comment, marked as answer]
@NBA707 why torture yourself so? Just use the full names 🙂 I personally much prefer to read `inventory.stock.location.checker` over `inv.stk.loc.checker`. Like is `inv` inventory or invoice for example - using the fullname ensures it's completely clear.

I regards to whether that folder structure is correct (mapping to the navbar), only you can decide if that feels right for your app.

I do generally like to group things by business domain (customers, inventory, etc) and maybe you need a level or two of sub organisation. But I would only do that if the top level business domain folder is starting to get hard to manage, otherwise I would keep it simple and less folders as long as possible. Like `inventory.stock-location-checker` for example.

Hope this helps!

**Reply by NBA707** [reply]
That's what I think of as torture.   I can only type what?  A trillion characters in my life time.... why waste them on super long unneccessary naming conventions?  No, no I am not stuck in the 90's when that was common place :)

Appreciate the feedback.    Naming convention asside, what about directory structure?   What kind of structure do you enjoy using when you are dealing with 40+ topics?

**Reply by joshhanley** [reply]
@NBA707 heh yeah fair! I like to group folders by domain and use suffixes to distinguish the different types of components within a directory, like Page for page components, Component for general components that should be inside other components, Form for form objects, etc.

Mostly my directories are pretty flat though. You can see an example of that here in the video from my recent LaraconAU talk https://youtu.be/esD6o8HYV5Q?t=226

**Comment by NBA707** [comment]
Yes, inventory is called stock in the system invc is for invoices.   So bad example.   Hard to break old habbits.  When I was working we were limited to 8 ccharacter long directories and 8 character long field names.   Not that I don't enjoy typing but keep it short was the rule of that day.

Should the directory structure of livewire components follow the menu or follow the major topic and minor topics?   Granted almost all topcis interreact with each other.   The menu can change over time so that just seems wrong to me.

Will go with topic based folder structures for now.

**Comment by NBA707** [comment]
I think the directory structure is more what my question was about than naming conventions.   Writting clean code does require a certain level of veboseness in the naming of things.   For me I dislike the Larvel structure and would like to have controllers, views, components closer to eachother in the directory structure, and be able to have a directory structure that groups things in a logical order.   Kind of like having a business complex, individual businesses, and inside the business have different offices or cubicals that deal with specific things.    That way I don't have to jump all around or rely on VS Code to know how to open the other thingy that needs TLC.

I like the idea of business domain, and sub level of key topics, and if those topics get too bloated sub topics.

**Reply by vitalijalbu** [reply]
how much data will you handle?

**Reply by NBA707** [reply]
It's about 500GB for primary data, and about 2 TB of secondary data.

**Reply by NBA707** [reply]
Typical business ERP system.   Complete GL / Accounting, Purchase Orders, Sales Orders, Invoices, Work Packages, Customer Quotes, Vendor Quotes, Inventory, Manuals, etc...

**Reply by vitalijalbu** [reply]
wow nice, do you use filamentphp?

**Reply by NBA707** [reply]
Filamentphp was not around when I started this project.  It's Laravel + Boostrap 3 & Bootstrap 5 & jQuery with a sprinkling on Livewire and things like DataTables.net & Select 2.

There are 14,535 directories and 161,162 files.   Some pretty monolithic controllers, some rare small controllers, lots of models,

**Reply by NBA707** [reply]
I have 1,196 lines in my web.php route.   Granted lots of comments to help group routes and a blank line or two....

**Reply by NBA707** [reply]
Have not leared TailwindCSS I see it as more of a pain than a solution.   Short and concise is what I preferr and pre-Bootstrap 5 it had an enourmase value no doubt.   Just don't have time to write all the components I use in Bootstrap in Tailwind CSS, granted there is that cool tool (Wendy) that would convert them.   Perhaps it wouldn't be that bad.
